chronos/transports/datagram.nim

```diff
@@ -601,7 +631,12 @@ proc close*(transp: DatagramTransport) =
   else:
     if {ReadClosed, WriteClosed} * transp.state == {}:
       transp.state.incl({WriteClosed, ReadClosed})
-      closeSocket(transp.fd, continuation)
+      try:
+        closeSocket(transp.fd, continuation)
+      except IOSelectorsException as exc:
+        raiseAssert exc.msg
```

Comment:
This is absolutely possible that `close` could raise, and it should not be `Defect`.

Reply:
well, this is the tricky part: `closeSocket` will raise only if the internal state of the ioselector is inconsistent - `closeSocket` raises when an invalid `fd` is passed to it - later, it raises an unrecoverable error in `unregister` as well, but that happens in a continuation which can no longer be caught _here_

Reply:
`closeSocket` could raise `EBADF`, `EINTR` (not handled right now), `EIO` (i'm talking about it).

Reply:
going through the code, it turns closeSocket doesn't raise in its present form but perhaps should - that's outside the scope of this PR however